pkg/chain/ethereum/ethereum.go

```diff
@@ -237,7 +244,42 @@ func (ec *ethereumChain) SubmitRelayEntry(
 		}
 	}()
 
-	_, err = ec.keepRandomBeaconOperatorContract.RelayEntry(entry)
+	transactorOptions := bind.NewKeyedTransactor(ec.accountKey.PrivateKey)
+	from := transactorOptions.From
+
+	toAddress, err := ec.config.ContractAddress("KeepRandomBeaconOperator")
+	if err != nil {
+		logger.Errorf("failed to find KeepRandomBeaconOperator address [%v]", err)
+	}
+
+	abi, err := goethabi.JSON(strings.NewReader(keepcoreabi.KeepRandomBeaconOperatorABI))
+	if err != nil {
+		logger.Errorf("failed to instantiate ABI: [%v]", err)
+	}
+
+	packed, err := abi.Pack("relayEntry", entry)
+	if err != nil {
+		logger.Errorf("failed packing relayEntry to conform with ABI [%v]", err)
+	}
+
+	msg := ethereum.CallMsg{
+		From: from,
+		To:   toAddress,
+		Data: packed,
+	}
+
+	gasEstimate, err := ec.client.EstimateGas(context.TODO(), msg)
+	if err != nil {
+		logger.Errorf("failed to estimate gas for relay entry [%v]", err)
+	}
```

Comment:
All of it should perhaps be a generated code inside `contract.KeepRandomBeaconOperator`. We can generate `*EstimateGas` function for each transactional function we generate in `keep-commons`. Then, we'd have `RelayEntryGasEstimate`.

@Shadowfiend What do you think about it?

Reply:
[PR](https://github.com/keep-network/keep-common/pulls) in `keep-commons`